doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md

```diff
@@ -150,12 +150,66 @@ The `newClient()` static method receives two mandatory arguments -- the platform
+
+<ul class="nav nav-tabs">
+	<li class="active"><a data-toggle="tab" href="#not-empty-constructor">DesktopKaaPlatformContext not empty constructor</a></li>
+	<li><a data-toggle="tab" href="#builder">FlexibleExecutorContext builder</a></li>
+	<li><a data-toggle="tab" href="#simpleExecutorContext">SimpleExecutorContext not empty constructor</a></li>
+</ul>
+
+
+<div class="tab-content">
+<div id="not-empty-constructor" class="tab-pane fade in active" markdown="1">
+
+```java
+KaaClientProperties properties = new KaaClientProperties();
+properties.put(CUSTOM_PROPERTY_NAME, CUSTOM_PROPERTY_VALUE);
+new DesktopKaaPlatformContext(properties,
+        CUSTOM_MAX_LIFE_CYCLE_THREADS, CUSTOM_MAX_API_THREADS,
+        CUSTOM_MAX_CALLBACK_THREADS, CUSTOM_MIN_SCHEDULED_THREADS
+    );
+```
+</div>
+
+<div id="simpleExecutorContext" class="tab-pane fade" markdown="1">
+
+```java
+ExecutorContext executor = new SimpleExecutorContext(
+        CUSTOM_MAX_LIFE_CYCLE_THREADS, CUSTOM_MAX_API_THREADS,
+        CUSTOM_MAX_CALLBACK_THREADS, CUSTOM_MIN_SCHEDULED_THREADS
+    );
+    
+new DesktopKaaPlatformContext();
+```
+</div>
+
+<div id="builder" class="tab-pane fade" markdown="1">
+
+```java
+ExecutorContext executorContext = new FlexibleExecutorContext.FlexibleExecutorContextBuilder()
+        .setMaxLifeCycleThreads(CUSTOM_MAX_LIFE_CYCLE_THREADS) // set custom maximum life cycle threads amount
+        .setMaxLifeCycleThreadsIdleMilliseconds(CUSTOM_MAX_LIFECYCLE_THREADS_IDLE_MILLISECONDS)
+        // set custom maximum life cycle threads idle time
+        .setMaxApiThreads(CUSTOM_MAX_API_THREADS) // set custom maximum api threads amount
+        .setMaxApiThreadsIdleMilliseconds(CUSTOM_MAX_API_THREADS_IDLE_MILLISECONDS) // set custom maximum api threads idle time
+        .setMaxCallbackThreads(CUSTOM_MAX_CALLBACK_THREADS) // set custom maximum callback threads amount
+        .setMaxCallbackThreadsIdleMilliseconds(CUSTOM_MAX_CALLBACK_THREADS_IDLE_MILLISECONDS)//set custom maximum callback threads idle time
+        .setMinScheduledThreads(CUSTOM_MIN_SCHEDULED_THREADS) // set custom minimum scheduled threads amount
+        .build();
+new DesktopKaaPlatformContext(null, executorContext);
+```
+</div>
+
+</div>
+
+In `KaaClient` creation example, the second argument is the default implementation of `KaaClientStateListener` -- `SimpleKaaClientStateListener`, that is solely used to log the client state changes.
```

Comment:
Please **return** this paragraph in position before ExecutorContext description.
It must be close to example which it describes.